src/agent/src/main.rs

```diff
@@ -187,6 +187,13 @@ fn main() -> Result<()> {
 
     announce(&logger, &config);
 
+    if args.len() == 2 && args[1] == "--version" {
```

Comment:
This chunk looks wrong - we already handle `--version` above at line 106...?

Reply:
It may be a conflict-resolving issue... :-), I will fix it.